lib/types.ts

```diff
@@ -17,7 +17,7 @@ export type DeepPartial<T> = {
 /** Like Required but recursive */
 export type DeepRequired<T> = T extends Primitive
   ? NonNullable<T>
-  : T extends any[]
+  : T extends (any[] | ReadonlyArray<any>)
```

Comment:
isn't ReadonlyArray<any> good enough?

Reply:
Technically it is, but I think this way is better for two reasons:
1) It's more explicit code: it's unclear until you think for some time why a normal array can be matched by ReadonlyArray\<any>. Most people reading this code will wonder "where are they matching normal arrays?"
2) It's more future-proof. If in the future any[] and ReadonlyArray\<any> don't match the same set of types, this construction will ensure both are caught.

Reply:
Aaaand just realized I had made the change in the wrong place :| Fixed it.